storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -4010,6 +3994,30 @@ dberr_t fil_space_t::garbage_collect(bool shutdown)
     return err;
   }
 
+  /* Remove cached undo log segments in the rollback segment
+  header page which exist in system tablespace */
+  mtr.start();
+  mtr.x_lock_space(fil_system.sys_space);
+  for (trx_rseg_t &rseg : trx_sys.rseg_array)
+  {
+    if (rseg.space == fil_system.sys_space)
+    {
+      buf_block_t *block=
+        buf_page_get_gen(page_id_t{0, rseg.page_no}, 0,
+                         RW_X_LATCH, nullptr, BUF_GET, &mtr, &err);
+      if (!block)
+      {
+        mtr.commit();
+        return err;
+      }
+
+      mtr.memset(block, TRX_RSEG_UNDO_SLOTS + TRX_RSEG,
+                 TRX_RSEG_N_SLOTS * 4, 0xff);
```

Comment:
The 4 needs to be written as `TRX_RSEG_SLOT_SIZE`. I think that this needs a source code comment. Why is it necessary to overwrite this array? If this was linking to some undo pages, how and where will that garbage be collected?
We might also avoid this operation if it is a no-op, but I think that the size of the `MEMSET` record is rather small.